lib/rest.js

```diff
@@ -36,40 +36,64 @@ module.exports = class Rest {
             this.options.requestAttempts
         );
     }
-    isTransactionalMessageApi(url) {
+    /**
+     * helper for {@link getBulk} to determine if the url is a transactional message API
+     * @private
+     * @param {string} url
+     * @returns {boolean} true if the url is a transactional message API
+     */
+    _isTransactionalMessageApi(url) {
         return url && this.transactionalApis.some((api) => url.includes(api));
     }
     /**
      * Method that makes paginated GET API Requests using $pageSize and $page parameters
      *
      * @param {string} url of the resource to retrieve
      * @param {number} [pageSize] of the response, defaults to 50
+     * @param {string} [iteratorField] attribute of the response to iterate over ('items'|'definitions'|'entry')
      * @returns {Promise<object>} API response combined items
      */
-    async getBulk(url, pageSize) {
-        let iteratorField;
+    async getBulk(url, pageSize, iteratorField) {
         let page = 1;
         const baseUrl = url.split('?')[0];
-        const isTransactionalMessageApi = this.isTransactionalMessageApi(baseUrl);
+        const isTransactionalMessageApi = this._isTransactionalMessageApi(baseUrl);
+        const isLegacyApi = baseUrl && baseUrl.startsWith('/legacy/v1/');
         const queryParameters = new URLSearchParams(url.split('?')[1]);
         let collector;
         let shouldPaginate = false;
-        queryParameters.set('$pageSize', Number(pageSize || 50).toString());
+        let pageSizeName = '$pageSize';
+        let pageName = '$page';
+        let countName = 'count';
+        if (isLegacyApi) {
+            pageSizeName = '$top';
+            pageName = '$skip';
+            countName = 'totalResults';
+            page = 0; // legacy index starts with 0
+            if (pageSize != 50) {
+                // values other than 50 are ignored by at least some of the sub-endpoints; while others have 50 as the maximum.
+                pageSize = 50;
+            }
+        }
+        queryParameters.set(pageSizeName, Number(pageSize || 50).toString());
         do {
-            queryParameters.set('$page', Number(page).toString());
+            queryParameters.set(pageName, Number(page).toString());
             const responseBatch = await this._apiRequest(
                 {
                     method: 'GET',
                     url: baseUrl + '?' + decodeURIComponent(queryParameters.toString()),
                 },
                 this.options.requestAttempts
             );
-            if (Array.isArray(responseBatch.items)) {
-                iteratorField = 'items';
-            } else if (Array.isArray(responseBatch.definitions)) {
-                iteratorField = 'definitions';
-            } else {
-                throw new TypeError('Could not find an array to iterate over');
+            if (!iteratorField) {
```

Comment:
two parts 
- not sure we want this "auto select" and ability to pass because optional params makes it less clear what is expected.
- this way we are checking the values in the response to ensure it is actually an array, but if you pass an iterator, it doesnt check if the key actually exists. i would rather we align it.

Reply:
on 1) it makes mcdev a little less dependent on you releasing an update to sfmc-sdk in case i find another iterator. I did update the jsdoc to make the "optional" part of this parameter even clearer.

on 2) i changed the logic to accomodate for that but I don't really like an empty if-check... lemme know.